src/odemis/driver/smartpod.py

```diff
@@ -0,0 +1,840 @@
+
+        self.position = model.VigilantAttribute({}, readonly=True)
+
+        # will take care of executing axis move asynchronously
+        self._executor = CancellableThreadPoolExecutor(1)  # one task at a time
+
+        # Check referencing, and reference if not referenced
+        self.referenced = model.VigilantAttribute(axes_ref, readonly=True)
+        referenced = c_int()
+        self.core.Smarpod_IsReferenced(self._id, byref(referenced))
+        if not referenced.value:
+            logging.debug("SmartPod is not referenced. The device will not function until referencing occurs.")
+        else:
+            logging.debug("SmartPod is referenced")
+
+        self._speed = self.GetSpeed()
+        self._accel = self.GetAcceleration()
+        try:
+            self._updatePosition()    # will fail if not referenced.
+        except SmartPodError as ex:
+            if ex.errno == SmartPodDLL.SMARPOD_NOT_REFERENCED_ERROR:
+                logging.info("Must reference SmartPod controller before use.")
+            else:
+                raise
+
+    def terminate(self):
+        # should be safe to close the device multiple times if terminate is called more than once.
+        self.core.Smarpod_Close(self._id)
+        super(SmartPod, self).terminate()
+        
+    def GetSwVersion(self):
+        """
+        Request the software version from the DLL file
+        """
+        major = c_uint()
+        minor = c_uint()
+        update = c_uint()
+        self.core.Smarpod_GetDLLVersion(byref(major), byref(minor), byref(update))
+        ver = "%u.%u.%u" % (major.value, minor.value, update.value)
+        return ver
+
+    def IsReferenced(self):
+        """
+        Ask the controller if it is referenced
+        """
+        referenced = c_int()
+        self.core.Smarpod_IsReferenced(self._id, byref(referenced))
+        return bool(referenced.value)
+        
+    def GetMoveStatus(self):
+        """
+        Gets the move status from the controller.
+        Returns:
+            SmartPodDLL.SMARPOD_MOVING is returned if moving
+            SmartPodDLL.SMARPOD_STOPPED when stopped
+            SmartPodDLL.SMARPOD_HOLDING when holding between moves
+            SmartPodDLL.SMARPOD_CALIBRATING when calibrating
+            SmartPodDLL.SMARPOD_REFERENCING when referencing
+            SmartPodDLL.SMARPOD_STANDBY
+        """
+        status = c_uint()
+        self.core.Smarpod_GetMoveStatus(self._id, byref(status))
+        return status
+
+    def Move(self, pos):
+        """
+        Move to pose command. Non-blocking
+        pos: (dict str -> float): axis name -> position
+            This is converted to the pose C-struct which is sent to the SmartPod DLL
+        Raises: SmartPodError if a problem occurs
+        """
+        # convert into a smartpad pose
+        newPose = dict_to_pose(pos)
+
+        self.core.Smarpod_Move(self._id, byref(newPose), c_uint(1000), c_int(0))
```

Comment:
Please document the 1000 (ms holding) and 0 (non-blocking).
How about using HOLDTIME_INFINITE instead of 1s? We need to find out what happens when it's not holding (does it move due to gravity?), and when it holds (does it vibrate due to closed-loop?).